internal/core/metadata/v2/application/device.go

```diff
@@ -130,3 +131,45 @@ func DeviceNameExists(name string, dic *di.Container) (exists bool, err errors.E
 	}
 	return exists, nil
 }
+
+// PatchDevice executes the PATCH operation with the device DTO to replace the old data
+func PatchDevice(dto dtos.UpdateDevice, ctx context.Context, dic *di.Container) errors.EdgeX {
+	dbClient := v2MetadataContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+
+	var device models.Device
+	var edgeXerr errors.EdgeX
+	if dto.Id != nil {
+		device, edgeXerr = dbClient.DeviceById(*dto.Id)
```

Comment:
if id != nil, you should check id is not empty and it is an uuid

Reply:
Added validation and test cases for id is invalid and id is empty string.